developer_tests/namelist/test_parse_variables.f90

```diff
@@ -0,0 +1,81 @@
+! DART software - Copyright UCAR. This open source software is provided
+! by UCAR, "as is", without charge, subject to all terms of use at
+! http://www.image.ucar.edu/DAReS/DART/DART_download
+
+program test_parse_variables
+
+use        utilities_mod, only : find_namelist_in_file, check_namelist_read 
+use    mpi_utilities_mod, only : initialize_mpi_utilities, finalize_mpi_utilities
+use            types_mod, only : vtablenamelength, MISSING_R8
+use    default_model_mod, only : parse_variables_clamp, parse_variables, &
+                                 state_var_type, &
+                                 MAX_STATE_VARIABLE_FIELDS_CLAMP, &
+                                 MAX_STATE_VARIABLE_FIELDS
+
+use test ! fortran-testanything
+
+implicit none
+
+integer :: iunit, io
+type(state_var_type) :: state_vars, state_vars_clamp
+
+character(len=vtablenamelength) :: state_variables(MAX_STATE_VARIABLE_FIELDS) = ' '
+character(len=vtablenamelength) :: state_variables_clamp(MAX_STATE_VARIABLE_FIELDS_CLAMP) = ' '
+
+namelist /model_nml/  &
+   state_variables
+
+namelist /model_nml_clamp/  &
+   state_variables_clamp
+
+call initialize_mpi_utilities('test_parse_variables')
+
+call plan(26)
+
+! Using namelist entry WITHOUT clamping values
+
+call find_namelist_in_file('input.nml', 'model_nml', iunit)
+read(iunit, nml = model_nml, iostat = io)
+call check_namelist_read(iunit, io, 'model_nml')
+
+state_vars = parse_variables(state_variables)
+
+call ok(state_vars%nvars == 3)
+call ok(state_vars%netcdf_var_names(1) == 'SALT_CUR')
+call ok(state_vars%netcdf_var_names(2) == 'TEMP_CUR')
+call ok(state_vars%netcdf_var_names(3) == 'UVEL_CUR')
+call ok(state_vars%qtys(1) == 263)
+call ok(state_vars%qtys(2) == 254)
+call ok(state_vars%qtys(3) == 356)
```

Comment:
This hardcoding the QTY index, 
The QTY actual number is created by preprocess, from the qty files. 
So this will break if someone adds, for example, a new land qty

assimilation_code/modules/observations/land_quantities_mod.f90

! BEGIN DART PREPROCESS QUANTITY DEFINITIONS
!
!   QTY_BOURBON_BISCUIT <- the QTY numbers 263, 254, 356 will no longer match the QTY_strings you have
!   QTY_ABSORBED_PAR

Reply:
fixed in e5057dadd609099e7405c9093fcb6baa39cc24f0